kannon/master.py

```diff
@@ -85,19 +85,25 @@ def build(self, root_task: gokart.TaskOnKart) -> None:
 
     def _create_task_queue(self, root_task: gokart.TaskOnKart) -> Deque[gokart.TaskOnKart]:
         task_queue: Deque[gokart.TaskOnKart] = deque()
+        visited_task_ids: Set[str] = set()
```

Comment:
[comment]

`deque` object has `__contains__` method so we can directly check existance

```python
>>> from collections import deque
>>> d = deque(range(0, 5))
>>> d
deque([0, 1, 2, 3, 4])
>>> 2 in d
True
>>> 6 in d
False
```

In my understanding, however, deque is implemented as linked list, lookup is not efficient, O(n).

So I agree with current implementation.

----

[imo] We can encapsulated "deque with quick lookup" into a class like

```
class EfficientDeque:
    queue: Deque[gokart.TaskOnKart] = deque()
    items: Set[str] = set()

    def append(item):
        self.queue.append(item)

    def popleft():
        return self.queue.popleft()

    def __contains__(item):
        return item in self.items
```

but could be too much ...?

Reply:
> [imo] We can encapsulated "deque with quick lookup" into a class like

Thanks to the proposal, but I think it is not required because it seems not general one.